db/external_sst_file_ingestion_job.cc

```diff
@@ -744,10 +717,96 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
   } else {
     return Status::InvalidArgument("External file version is not supported");
   }
+
+  file_to_ingest->cf_id = static_cast<uint32_t>(props->column_family_id);
+  // This assignment works fine even though `table_reader` may later be reset,
+  // since that will not affect how table properties are parsed, and this
+  // assignment is making a copy.
+  file_to_ingest->table_properties = *props;
+
   // Get number of entries in table
   file_to_ingest->num_entries = props->num_entries;
   file_to_ingest->num_range_deletions = props->num_range_deletions;
 
+  // Validate table properties related to comparator name and user defined
+  // timestamps persisted flag.
+  const bool orig_udt_persisted =
```

Comment:
Do we need this field `file_to_ingest->user_defined_timestamps_persisted`? It seems it's used only in this function and `orig_udt_persisted` is always true.

Reply:
Yes, we use this field main for two places:
1) Pass this field to create the `TableReader`.
2) Pass this field to create the `FileMetaData` in the file's `VersionEdit`

So I added this field to `IngestedFileInfo` because it's needed in both `Prepare` and `Run`.

Reply:
Just want to add some notes about `orig_udt_persisted`. Yes, the way we currently initialize the default value for `IngestedFileInfo.user_defined_timestamps_persisted`, it's always true. So strictly speaking, here we don't need to specifically track the original udt persisted value, we just need to check the new udt persisted value, and re-create the `TableReader` if it's false.

There is however implicit dependency between how we initialize that flag's default value with how we handle re-creating a `TableReader`, which I think shouldn't be the case. It will cause error if the default value is accidently updated to be defaulted to false. So a more reliable way is to strictly tracking if the value of `IngestedFileInfo.user_defined_timestamps_persisted` flag is updated after this sanity check.

Reply:
I missed the one in Run(). It makes sense that we need to set it in Prepare(). However, I feel we don't need to use its value to create the initial TableReader to create the dependency. We can make `orig_udt_persisted` a global const flag and use it directly to create the first TableReader?

Reply:
That's a good point! One of the side effect of having this flag in `IngestedFileInfo` is that I can keep the logic to create the `TableReader` in function `ResetTableReader` consistently use the `IngestedFileInfo.user_defined_timestamps_persisted` flag.

With this global const flag approach, we just need to add some logic to track if this is the first time the `TableReader` is being created.  I will try this out.

Reply:
You can also pass in an additional parameter to `ResetTableReader`.